entities/src/main/java/com/flipkart/varadhi/entities/VaradhiTopic.java

```diff
@@ -43,11 +41,14 @@ public void addInternalTopic(InternalTopic internalTopic) {
         this.internalTopics.put(internalTopic.getTopicRegion(), internalTopic);
     }
 
-    public InternalTopic getProduceTopicForRegion(String region) {
-        InternalTopic internalTopic = internalTopics.get(region);
-        if (null == internalTopic) {
-            throw new ResourceNotFoundException(String.format("Topic not found for region(%s).", region));
-        }
-        return internalTopic;
+    /**
+     * TODO: evaluate whether it should return Optional or not. An unmapped region probably wouldn't be queried.
```

Comment:
agree, unmapped region shouldn't be queried. If it does' letting it fail is better.
Optional seems additional overhead, as caller need to do the additional validations on top of it, so can be removed.